frontend/src/concepts/topology/PipelineVisualizationSurface.tsx

```diff
@@ -25,14 +26,37 @@ import { NODE_HEIGHT, NODE_WIDTH } from './const';
 type PipelineVisualizationSurfaceProps = {
   nodes: PipelineNodeModel[];
   selectedIds?: string[];
+  sidePanel?: React.ReactElement | null;
 };
 
 const PipelineVisualizationSurface: React.FC<PipelineVisualizationSurfaceProps> = ({
   nodes,
   selectedIds,
+  sidePanel,
 }) => {
   const controller = useVisualizationController();
   const [error, setError] = React.useState<Error | null>();
+
+  React.useEffect(() => {
+    let resizeTimeout: NodeJS.Timeout | null;
+
+    if (selectedIds?.[0]) {
+      const selectedNode = controller.getNodeById(selectedIds[0]);
+      if (selectedNode) {
+        // Use a timeout in order to allow the side panel to be shown and window size recomputed
+        resizeTimeout = setTimeout(() => {
+          controller.getGraph().panIntoView(selectedNode, { offset: 20, minimumVisible: 100 });
+          resizeTimeout = null;
+        }, 500);
```

Comment:
it looks like there is a callback we can use in the `TopologyView`

```tsx
  /** Callback for side bar resize end. */
  onSideBarResize?: (width: number, id: string) => void;
```

Reply:
We found that this callback only fires when the sidebar is resized manually and not on the initial render